### PR TITLE
range-based loops and const qualifications in net.cpp

### PR DESCRIPTION
**Pull request by benma**
Plus a use of std::copy() instead of manual copying.

(The loop on line 117 is already done in #10493).

### CODE REVIEW

File: src/net.cpp [hunk 1]
```diff
@@ -305,16 +305,16 @@ CNode* CConnman::FindNode(const CNetAddr& ip)
...
     return NULL;
 }
 
 CNode* CConnman::FindNode(const CSubNet& subNet)
 {
     LOCK(cs_vNodes);
     for (CNode* pnode : vNodes)
-    if (subNet.Match((CNetAddr)pnode->addr))
-        return (pnode);
+        if (subNet.Match((CNetAddr)pnode->addr))
```
Comment (jonasschnelli) [inline src/net.cpp]: Please switch to brackets here.

**Comment by practicalswift** [comment]
Concept ACK (strong ACK! :-))

File: src/net.cpp (hunk 1 above)
Comment (benma) [inline src/net.cpp]: Done.

**Review by dcousens — APPROVED** [review]

**Comment by corebob** [comment]
How about "improving" those FindNode functions to something like this,

```
auto iter = std::find_if(vNodes.begin(), vNodes.end(), [&](CNode *node) {
        return node->GetAddrName() == addrName;
    });

    return iter == vNodes.end() ? NULL : *iter;
```

 I believe C++11 allows this

**Comment by practicalswift** [comment]
@corebob I guess it is a matter of taste, but personally I find the current range-based for loop form more straightforward and easier to read:

```c++
     for (CNode* pnode : vNodes) {
          if (pnode->GetAddrName() == addrName) {
              return pnode;
          }
      }
      return NULL;
```

**Comment by corebob** [comment]
@practicalswift 
I pretty much agree. I think pro arguments could be made for getting rid of a raw loop, and getting rid of an extra exit point. Though probably not worth it in this case.

**Comment by benma** [comment]
I am usually in favor of using `algorithm`, as explicit names (`find_if`) and lambdas are more descriptive than loops etc., but in this case, I don't like the return value conversion, so it indeed looks more complicated than before. Too bad.

**Comment by benma** [comment]
Rebased.

**Comment by benma** [comment]
Can someone please retrigger CI?

**Comment by sipa** [comment]
utACK a9144d535122a4d08348e477315b5a94368e6dfc

Travis passed after restart.

File: src/net.cpp [hunk 2]
```diff
@@ -135,11 +135,11 @@ static std::vector<CAddress> convertSeed6(const std::vector<SeedSpec6> &vSeedsIn
     const int64_t nOneWeek = 7*24*60*60;
     std::vector<CAddress> vSeedsOut;
     vSeedsOut.reserve(vSeedsIn.size());
-    for (std::vector<SeedSpec6>::const_iterator i(vSeedsIn.begin()); i != vSeedsIn.end(); ++i)
+    for (const auto seed_in : vSeedsIn)
```
Comment (MarcoFalke) [inline src/net.cpp]: `const SeedSpec6& seed_in` makes more sense to me as it prevents a copy, no?

**Review by maflcko — COMMENTED** [review]
utACK a9144d535.

Since this is refactoring only you also might want to use clang format on the changed lines. (See the developer notes for a one-liner to fixup the formatting)

File: src/net.cpp (hunk 2 above)
Comment (benma) [reply src/net.cpp]: Correct. Done!

**Comment by benma** [comment]
@MarcoFalke I also applied clang-format. Nice tool. Maybe it should go into the CI for all future changes ;)

**Comment by MarcoFalke** [comment]
re-utACK 05cae8a

**Comment by benma** [comment]
Can this be merged?

**Pull request merged**